servers/rendering/renderer_rd/effects/ss_effects.cpp

```diff
@@ -1079,79 +1007,45 @@ void SSEffects::ssao_allocate_buffers(SSAORenderBuffers &p_ssao_buffers, const S
 
-		{
-			RD::TextureFormat tf;
-			tf.format = RD::DATA_FORMAT_R8G8_UNORM;
-			tf.texture_type = RD::TEXTURE_TYPE_2D_ARRAY;
-			tf.width = p_ssao_buffers.buffer_width;
-			tf.height = p_ssao_buffers.buffer_height;
-			tf.array_layers = 4;
-			tf.usage_bits = RD::TEXTURE_USAGE_SAMPLING_BIT | RD::TEXTURE_USAGE_STORAGE_BIT;
-			p_ssao_buffers.ao_pong = RD::get_singleton()->texture_create(tf, RD::TextureView());
-			RD::get_singleton()->set_resource_name(p_ssao_buffers.ao_pong, "SSAO De-interleaved Array Pong");
-			for (uint32_t i = 0; i < 4; i++) {
-				RID slice = RD::get_singleton()->texture_create_shared_from_slice(RD::TextureView(), p_ssao_buffers.ao_pong, i, 0);
-				p_ssao_buffers.ao_pong_slices.push_back(slice);
-				RD::get_singleton()->set_resource_name(slice, "SSAO De-interleaved Array Layer " + itos(i) + " Pong");
-			}
-		}
+	// As we're not clearing these, and render buffers will return the cached texture if it already exists,
+	// we don't first check has_texture here
 
-		{
-			RD::TextureFormat tf;
-			tf.format = RD::DATA_FORMAT_R8_UNORM;
-			tf.width = p_ssao_buffers.buffer_width;
-			tf.height = p_ssao_buffers.buffer_height;
-			tf.usage_bits = RD::TEXTURE_USAGE_SAMPLING_BIT | RD::TEXTURE_USAGE_STORAGE_BIT;
-			p_ssao_buffers.importance_map[0] = RD::get_singleton()->texture_create(tf, RD::TextureView());
-			RD::get_singleton()->set_resource_name(p_ssao_buffers.importance_map[0], "SSAO Importance Map");
-			p_ssao_buffers.importance_map[1] = RD::get_singleton()->texture_create(tf, RD::TextureView());
-			RD::get_singleton()->set_resource_name(p_ssao_buffers.importance_map[1], "SSAO Importance Map Pong");
-		}
-		{
-			RD::TextureFormat tf;
-			tf.format = RD::DATA_FORMAT_R8_UNORM;
-			tf.width = p_settings.full_screen_size.x;
-			tf.height = p_settings.full_screen_size.y;
-			tf.usage_bits = RD::TEXTURE_USAGE_SAMPLING_BIT | RD::TEXTURE_USAGE_STORAGE_BIT;
-			p_ssao_buffers.ao_final = RD::get_singleton()->texture_create(tf, RD::TextureView());
-			RD::get_singleton()->set_resource_name(p_ssao_buffers.ao_final, "SSAO Final");
-		}
-		p_ssao_buffers.half_size = ssao_half_size;
-	}
+	p_render_buffers->create_texture(RB_SCOPE_SSAO, RB_DEINTERLEAVED, RD::DATA_FORMAT_R8G8_UNORM, RD::TEXTURE_USAGE_SAMPLING_BIT | RD::TEXTURE_USAGE_STORAGE_BIT, RD::TEXTURE_SAMPLES_1, full_size, 4 * view_count);
+	p_render_buffers->create_texture(RB_SCOPE_SSAO, RB_DEINTERLEAVED_PONG, RD::DATA_FORMAT_R8G8_UNORM, RD::TEXTURE_USAGE_SAMPLING_BIT | RD::TEXTURE_USAGE_STORAGE_BIT, RD::TEXTURE_SAMPLES_1, full_size, 4 * view_count);
+	p_render_buffers->create_texture(RB_SCOPE_SSAO, RB_IMPORTANCE_MAP, RD::DATA_FORMAT_R8_UNORM, RD::TEXTURE_USAGE_SAMPLING_BIT | RD::TEXTURE_USAGE_STORAGE_BIT, RD::TEXTURE_SAMPLES_1, full_size);
+	p_render_buffers->create_texture(RB_SCOPE_SSAO, RB_IMPORTANCE_PONG, RD::DATA_FORMAT_R8_UNORM, RD::TEXTURE_USAGE_SAMPLING_BIT | RD::TEXTURE_USAGE_STORAGE_BIT, RD::TEXTURE_SAMPLES_1, full_size);
```

Comment:
I feel like these should be ``half_size`` like they are for SSIL. I don't know why they are different. But the important map phase should be at 1/4 resolution instead of 1/2 (or 1/4 and 1/8 when using half size)

Reply:
Hmmm Ultra seems to have some very strange issues. Full size using Ultra takes about the same amount of time as half size very low quality for me.

Medium quality
![Screenshot from 2023-02-09 14-50-17](https://user-images.githubusercontent.com/16521339/217957127-579ceeba-c1ac-4d2c-a2f8-c35b4f477132.png)

Very low quality
![Screenshot from 2023-02-09 14-50-09](https://user-images.githubusercontent.com/16521339/217957133-8d385765-5008-4e38-b39f-a3a79eadea99.png)

Qltra quality
![Screenshot from 2023-02-09 14-50-03](https://user-images.githubusercontent.com/16521339/217957134-6136b33e-7a8d-430e-a9d6-ad754dd54d7b.png)

Reply:
> I feel like these should be half_size like they are for SSIL. I don't know why they are different. But the important map phase should be at 1/4 resolution instead of 1/2 (or 1/4 and 1/8 when using half size)

I think that must have been lost in the past. Looking at the code that fill the importance maps, it is indeed sending half size to the compute shader so I think you are right.

I've changed them, lets see what happens, it looks good for me so far.